### PR TITLE
Auto-detect single project

### PR DESCRIPTION
**Pull request by timotei**
Try to "guess" which is the project to be used
as the main project when we have no projects
which have the same name as the gradle project.

### CODE REVIEW

File: src/main/groovy/com/ullink/Msbuild.groovy [hunk 1]
```diff
@@ -107,6 +107,18 @@ class Msbuild extends ConventionTask {
                     parseProject = false
                 } else {
                     projectParsed = allProjects[projectName]
+
+                    if (projectParsed == null) {
+                        def potentialProject = allProjects.collect {it.value}.find {
```
Comment (gluck) [inline src/main/groovy/com/ullink/Msbuild.groovy]: allProjects.values()

File: src/main/groovy/com/ullink/Msbuild.groovy [hunk 2]
```diff
@@ -107,6 +107,18 @@ class Msbuild extends ConventionTask {
...
+                            it.properties.containsKey('ProjectDir')
```
Comment (gluck) [inline src/main/groovy/com/ullink/Msbuild.groovy]: so you consider the first project of solution is the main one ?

Comment (timotei) [inline src/main/groovy/com/ullink/Msbuild.groovy]: My first idea would be to detect if there is a *single* project and if not abort the automatic detection. Not sure if it's better to take the first one if there are more or not.

**Comment by timotei** [comment]
Any feedback on this?

File: build.gradle [hunk 3]
```diff
@@ -1,11 +1,10 @@
 plugins {
     id 'nu.studer.plugindev' version '1.0.3'
     id 'com.jfrog.bintray' version '1.2'
-    id 'com.ullink.msbuild' version '2.18'
-    id 'com.ullink.nuget' version '2.15'
+    id 'com.ullink.msbuild' version '2.12'
```
Comment (ngyukman) [inline build.gradle]: should not rollback those gradle plugin?
I think it could be the failure reason

Comment (timotei) [reply build.gradle]: Yes, was unintended

**Comment by timotei** [comment]
@gluck @ngyukman I had to add the archive with ProjectFileParser for the tests to work fine, not the best, but it's just as test resource

File: build.gradle [hunk 4]
```diff
@@ -41,7 +41,7 @@ msbuild {
 }
 
 nunit {
-    nunitVersion = '3.0.1'
+    nunitVersion = '3.8.0'
```
Comment (dfev77) [inline build.gradle]: If latest nunit plugin is referenced that would default the nunit console runner to 3.9.0

File: src/main/groovy/com/ullink/Msbuild.groovy [hunk 5]
```diff
@@ -127,9 +127,21 @@ class Msbuild extends ConventionTask {
                     parseProject = false
                 } else {
                     projectParsed = allProjects[projectName]
```
Comment (dfev77) [inline src/main/groovy/com/ullink/Msbuild.groovy]: Why would one access a project that does not exist and why would we try to guess why she should actually access instead

**Pull request merged**